Source/SwiftLintBuiltInRules/Rules/Idiomatic/UnusedEnumeratedRule.swift

```diff
@@ -95,41 +149,47 @@ private extension UnusedEnumeratedRule {
                     onePosition: firstTokenIsUnderscore ? nil : secondElement.positionAfterSkippingLeadingTrivia
                 )
             } else {
-                self.trailingClosure = trailingClosure
+                nextClosure = trailingClosure
             }
 
             return .visitChildren
         }
 
+        override func visit(_ node: ClosureExprSyntax) -> SyntaxVisitorContinueKind {
+            if node == nextClosure {
+                closures.append(Closure(closure: node))
+                nextClosure = nil
+            }
+            return .visitChildren
+        }
+
         override func visitPost(_ node: ClosureExprSyntax) {
-            guard node == trailingClosure else {
+            guard let closure = closures.last, node == closure.closure else {
                 return
             }
-            defer {
-                trailingClosure = nil
-                zeroPosition = nil
-                onePosition = nil
-            }
-            guard (zeroPosition != nil) != (onePosition != nil) else {
+            defer { closures.removeLast() }
+
+            guard (closure.zeroPosition != nil) != (closure.onePosition != nil) else {
                 return
             }
 
-            addViolation(zeroPosition: zeroPosition, onePosition: onePosition)
+            addViolation(zeroPosition: closure.zeroPosition, onePosition: closure.onePosition)
         }
 
         override func visitPost(_ node: DeclReferenceExprSyntax) {
-            guard trailingClosure != nil else {
+            guard var closure = closures.last, node.baseName.text == "$0" || node.baseName.text == "$1" else {
```

Comment:
How does it make sure that the closure is the right one?

Reply:
https://github.com/realm/SwiftLint/pull/5498/files#r1552625338